lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -62,15 +64,33 @@ sub open_connection {
 
     $c->on(text => sub {
         my ($c, $msg) = @_;
+
+        my $original = "$msg";
```

Comment:
Is the string interpolation intentional?

Reply:
yes so as to be double sure that nothing modifies it

Reply:
@zakame We want to avoid any corruption caused by XS code or other unexpected behaviour - interpolating is only a partial guard against this, we're likely to hit CoW as well, so we may need `'' . $msg` or something more aggressive if this isn't enough.

Reply:
would pay to add a  comment  with the above